File: PHP/src/DAO/ItemDao.php
<?php

class ItemDao {

    const _table = 'item';

    public function __construct() {
        
    }

    public function insert(Item $item) {
        $db = Database::singleton();

        $sql = 'INSERT INTO ' . self::_table . ' (name, description, amount) VALUES (?,?,?)';

        $sth = $db->prepare($sql);

        $sth->bindValue(1, strtolower(trim($item->name)), PDO::PARAM_STR);
        $sth->bindValue(2, trim($item->description), PDO::PARAM_STR);
        $sth->bindValue(3, trim($item->amount), PDO::PARAM_STR);

        return $sth->execute();
    }

    public function update(Item $item) {
        $db = Database::singleton();

        $sql = 'UPDATE ' . self::_table . ' SET name = ?, description = ?, amount = ?, last_update = now() WHERE id = ?';

        $sth = $db->prepare($sql);

        $sth->bindValue(1, strtolower(trim($item->name)), PDO::PARAM_STR);
        $sth->bindValue(2, trim($item->description), PDO::PARAM_STR);
        $sth->bindValue(3, $item->amount, PDO::PARAM_INT);
        $sth->bindValue(4, $item->id, PDO::PARAM_INT);

        return $sth->execute();
    }

    public function getAll() {

        $db = Database::singleton();

        $sql = 'SELECT * FROM ' . self::_table;

        $sth = $db->prepare($sql);

        $sth->execute();

        $itens = array();

        while ($obj = $sth->fetch(PDO::FETCH_OBJ)) {
            $item = new Item();
            
            $item->id = $obj->id;
            $item->name = $obj->name;
            $item->description = $obj->description;
            $item->amount = $obj->amount;
            $item->createdDate = $obj->created_date;
            $item->lastUpdate = $obj->last_update;
            
            array_push($itens, $item);
        }

        return $itens;
    }
    
    public function get($id){
        
        $db = Database::singleton();

        $sql = 'SELECT * FROM ' . self::_table . ' WHERE id = ?';

        $sth = $db->prepare($sql);
        
        $sth->bindValue(1, strtolower(trim($id)), PDO::PARAM_STR);
       
        $sth->execute();
        
        if($obj = $sth->fetch(PDO::FETCH_OBJ)){
            
            $item = new Item();
            
            $item->id = $obj->id;
            $item->name = $obj->name;
            $item->description = $obj->description;
            $item->amount = $obj->amount;
            
            return $item;
            
        }
        
        
    }

}

File: JavaWeb/src/java/dao/ItemDao.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package dao;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import java.util.List;
import model.Item;
import util.ConnectionFactory;

/**
 *
 * @author SoldierJVX
 */
public class ItemDao {

    private Connection connection;

    public ItemDao() {

        try {
            this.connection = ConnectionFactory.getConnection();
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public Item get(int id) {

        Item item = null;

        try {
            String sql = "SELECT * FROM item WHERE id = " + id;

            Statement stmt = this.connection.createStatement();

            ResultSet rs = stmt.executeQuery(sql);

            if (rs.next()) {
                item = new Item(rs.getInt("id"), rs.getString("name"), rs.getString("description"), rs.getInt("amount"), rs.getDate("created_date"), rs.getDate("last_update"));
            }

        } catch (SQLException e) {
            e.printStackTrace();
        }

        return item;

    }

    public List<Item> list() {

        List<Item> list = new ArrayList<>();

        try {
            String sql = "SELECT * FROM item ORDER BY id";

            Statement stmt = this.connection.createStatement();

            ResultSet rs = stmt.executeQuery(sql);

            while (rs.next()) {
                list.add(new Item(rs.getInt("id"), rs.getString("name"), rs.getString("description"), rs.getInt("amount"), rs.getDate("created_date"), rs.getDate("last_update")));
            }

            return list;

        } catch (SQLException e) {
            e.printStackTrace();
        }

        return list;

    }

    public boolean insertItem(String nome, String descricao, int quantidade) {

        try {
            String sql = "INSERT INTO item (name, description, amount) VALUES ('" + nome + "', '" + descricao + "', " + quantidade + ")";

            Statement stmt = this.connection.createStatement();

            stmt.executeUpdate(sql);

            return true;

        } catch (SQLException e) {
            e.printStackTrace();
        }

        return false;

    }

    public boolean updateItem(int id, String nome, String descricao, String quantidade) {
        try {
            String sql = "UPDATE item SET name = '" + nome + "', description = '" + descricao + "', amount = " + quantidade + ", last_update = now() WHERE id = " + id;

            Statement stmt = this.connection.createStatement();

            stmt.executeUpdate(sql);

            return true;

        } catch (SQLException e) {
            e.printStackTrace();
        }

        return false;
    }

}
